data/html-parser/index.ts

```diff
@@ -0,0 +1,83 @@
+// Essentially jQuery for parsing the DOM
+import cheerio from 'cheerio';
+import { defaults, pickAll } from 'lodash/fp';
+import DataTypes from '../types';
+import HtmlDataTypes from '../types/html';
+import { duplicateLangAttributes } from './duplicate-lang-attributes';
+import { liftLangAttributes } from './lift-lang-attributes';
+import { removeParagraphsFromDefinitionListsAndMerge } from './remove-paragraphs-from-definition-lists';
+
+type Attributes = { [attr: string]: string };
+
+type HtmlDataType = keyof typeof HtmlDataTypes;
+
+/**
+ * Data that we need to check from any cheerio element to see what to do with it
+ */
+type AnyCheerioElement = {
+  type?: null | undefined | 'text' | 'tag' | 'script' | 'style' | 'comment';
+  name: HtmlDataType;
+  data?: undefined | string;
+  attribs?: null | undefined | Attributes;
+  children: cheerio.Element[];
+};
+
+type ParsedNode = {
+  data: string | ParsedNode[];
+  type: typeof HtmlDataTypes.text | typeof HtmlDataTypes.tag;
+  name: string;
+  attribs?: null | undefined | Attributes;
+};
+
+export const cheerioNodeParser = (_i: number, node: cheerio.Element): ParsedNode => {
+  const {
+    type = null,
+    name = <HtmlDataType>HtmlDataTypes.text,
+    data = '',
```

Comment:
If the default is a value, is this data? or is the default value undefined?

Reply:
You're right, I was flailing trying to get Types to compile properly, I think this was an artefact of a previous attempt; have removed redundant defaults and all works fine.